### DISCUSSION TITLE
How to replace PDP with Simple Participant Discovery Protocol

### DISCUSSION THREAD

**Discussion by rookybird** [eProsima/Fast-DDS#4231] [Q&A]
Hi,
I am trying to implement static discovery, according to the [tutorial](https://fast-dds.docs.eprosima.com/en/latest/fastdds/discovery/discovery.html#disc-mechanisms:~:text=This%20mechanisms%20uses,are%20known%20beforehand.),  which uses Simple Participant Discovery Protocol (SPDP) and skips EDP with known entities in the xml file. 

I found examples on configuring [STATIC EDP](https://fast-dds.docs.eprosima.com/en/latest/fastdds/discovery/static.html#staticlocators:~:text=5.3.3.3.1.-,STATIC%20EDP%20XML%20Example,-The%20following%20is). But I didn't know how to set PDP to be SPDP. Do we need to configure it or is SPDP the default?

In the EDP example, I saw 
`pqos.wire_protocol().builtin.discovery_config.use_SIMPLE_EndpointDiscoveryProtocol = false;
pqos.wire_protocol().builtin.discovery_config.use_STATIC_EndpointDiscoveryProtocol = true;`

Do we also need to set the "discoveryProtocol" to be STATIC? like what we did for simple discovery [here](https://fast-dds.docs.eprosima.com/en/latest/fastdds/discovery/general_disc_settings.html#:~:text=%3D%0A%20%20%20%20%20%20%20%20DiscoveryProtocol_t%3A%3A-,SIMPLE,-%3B). But there is no option for STATIC.

Another question is how to calculate the port of unicast locator in static discovery, as indicated [here](https://fast-dds.docs.eprosima.com/en/latest/fastdds/discovery/static.html#staticlocators:~:text=192.168.0.128%22%20port%3D%22-,5000,-%22/%3E%0A%20%20%20%20%20%20%20%20%20%20%20%20%3CunicastLocator)

**Comment by JesusPoderoso** [comment]
Hi @rookybird thanks for using Fast DDS.
The [static discovery protocol](https://fast-dds.docs.eprosima.com/en/latest/fastdds/discovery/static.html) is configured through [static EDP](https://fast-dds.docs.eprosima.com/en/latest/fastdds/discovery/static.html#static-edp) only, there is no such static PDP.
You can configure the participant discovery statically by setting initial peers and disabling multicast, as explained in [this example](https://fast-dds.docs.eprosima.com/en/latest/fastdds/use_cases/wifi/disable_multicast.html).
